app/app.py

```diff
@@ -78,9 +87,16 @@ def find_recipe_in_url_window_algo_based():
     instructions = request.form['instructions'].lower() == "true"
     ingredients = request.form['ingredients'].lower() == "true"
     try:
+        if stored_recipes_dao.url_in_db(url):
+            return jsonify(stored_recipes_dao.get_recipe_from_db(url, ingredients, instructions))
```

Comment:
it is working I am not sure I understand what you mean.

if url is in db then the user will get the recipe from the db with the get_recipe_from_db function.
else it will do an extraction as usual and at the end will store the extracted recipe in the db.

Reply:
you right, got a bit confused cause github broke the line so I only see "return" disregard it.